tempto-core/src/main/java/com/teradata/tempto/internal/initialization/TestInitializationListener.java

```diff
@@ -112,22 +113,18 @@ public TestInitializationListener()
         return collectFulfillers(BUILTIN_SUITE_LEVEL_FULFILLERS, AutoSuiteLevelFulfiller.class);
     }
 
-    private static List<Class<? extends RequirementFulfiller>> collectFulfillers(List<Class<? extends RequirementFulfiller>> builtinFulfillers, Class<? extends Annotation> filterAnnotation)
+    // package scope due to testing
+    static List<Class<? extends RequirementFulfiller>> collectFulfillers(List<Class<? extends RequirementFulfiller>> builtinFulfillers, Class<? extends Annotation> filterAnnotation)
     {
-        ImmutableList.Builder<Class<? extends RequirementFulfiller>> resultFulfillers = ImmutableList.builder();
-        resultFulfillers.addAll(builtinFulfillers);
-        resultFulfillers.addAll(scanForFulfillersAndSort(filterAnnotation));
-
-        return resultFulfillers.build();
-    }
-
-    // package scope due testing
-    static List<Class<? extends RequirementFulfiller>> scanForFulfillersAndSort(Class<? extends Annotation> filterAnnotation)
-    {
-        return getAnnotatedSubTypesOf(RequirementFulfiller.class, filterAnnotation)
-                .stream()
-                .sorted(new RequirementFulfillerByPriorityComparator().reversed())
-                .collect(toList());
+        Set<Class<? extends RequirementFulfiller>> userFulfillers = getAnnotatedSubTypesOf(RequirementFulfiller.class, filterAnnotation);
+        List<Class<? extends RequirementFulfiller>> allFulfillers = new ArrayList<>(builtinFulfillers);
+        allFulfillers.addAll(userFulfillers);
+
+        return ImmutableList.copyOf(
+                allFulfillers.stream()
+                             .sorted(new RequirementFulfillerByPriorityComparator())
+                             .collect(toList())
```

Comment:
it would be nice to have toImmutableList collector. 
Airlift/Presto have one. If you have some time you may copy it from them :)